Classes/Backend/TceMain.php

```diff
@@ -535,9 +536,10 @@ public function moveRecord_afterAnotherElementPostProcess($table, $uid, $destPid
         // Following block takes care of updating the immediate record, be that a placeholder, an
         // original or a versioned copy.
         $moveData = $this->getMoveData();
-        if (!$moveData) {
-            return;
-        }
+
+        $resolveUid = $this->getOriginalRecordUid($table, $uid);
+        $updateFields['colPos'] = GeneralUtility::_GET('data')[$table][$resolveUid]['colPos'];
```

Comment:
Please do not access GET parameters unless there is absolutely no other possible way to do it. In this case, DataHandler ($reference) should contain the mapping array you access here.